api/v1/atlasdatabaseuser_types.go

```diff
@@ -197,6 +197,11 @@ func (p *AtlasDatabaseUser) ProjectDualRef() *ProjectDualReference {
 	return &p.Spec.ProjectDualReference
 }
 
+// IsDatabaseUserReady checks if the Ready condition is available
```

Comment:
note for future shelves: this will not play nice with autogenerated CRDs

But those use `metav1.Condition` so we might have a generic way instead of calling one per type like here.

Nothing actionable for this PR though